lib/block_view.go

```diff
@@ -2987,24 +3067,74 @@ func _isEntryImmatureBlockReward(utxoEntry *UtxoEntry, blockHeight uint32, param
 	return false
 }
 
-func _verifySignature(txn *MsgBitCloutTxn) error {
-	// Compute a hash of the transaction
+func (bav *UtxoView) _verifySignature(txn *MsgBitCloutTxn, blockHeight uint32) error {
+	// Compute a hash of the transaction.
 	txBytes, err := txn.ToBytes(true /*preSignature*/)
 	if err != nil {
 		return errors.Wrapf(err, "_verifySignature: Problem serializing txn without signature: ")
 	}
 	txHash := Sha256DoubleHash(txBytes)
-	// Convert the txn public key into a *btcec.PublicKey
-	txnPk, err := btcec.ParsePubKey(txn.PublicKey, btcec.S256())
+
+	// We make a local copy of the signature as it should be read-only.
+	sigCopy := append([]byte{}, txn.Signature...)
+	// Verify that the control byte is within the allowed range.
+	// DERControlByte means txn was signed with the owner's private key.
+	// Otherwise, txn was signed with a derived key and we will attempt
+	// to recover the public key from the signature.
+	sigByte := sigCopy[0]
```

Comment:
You have good comments but I'm still very confused by what's going on here. It looks like you're trying to encode a boolean isDerivedKey inside of the signature somehow. If this is the case, I think it's error-prone and a bit too clever. Instead, I would create a simple addition to extraData that contains this information, and modify this function to support an isDerivedKey boolean.